### ISSUE TITLE
Compile error with Recast.cpp for android template

### ISSUE THREAD

**Issue by volzhs** [godotengine/godot#11273]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Ubuntu 16.04.3, master branch 3f4062c

**Issue description:**

```
[ 90%] Compiling ==> thirdparty/recastnavigation/Recast/Source/Recast.cpp
thirdparty/recastnavigation/Recast/Source/Recast.cpp:26:10: fatal error: 'new' file not found
#include <new>
         ^~~~~
1 error generated.
scons: *** [thirdparty/recastnavigation/Recast/Source/Recast.android.opt.debug.armv7.neon.o] Error 1
scons: building terminated because of errors.
```

**Steps to reproduce:**
`scons platform=android tools=no` => fail
`scons platform=android tools=no android_stl=yes` => success

**Link to minimal example project:**

**Comment by akien-mga** [comment]
CC @SaracenOne

**Comment by hpvb** [comment]
Should recast be build at all for tools=no? I thought the point of the way we did this PR was that we'd only enable it for the editor?

**Comment by volzhs** [comment]
isn't that used for updating navmesh on runtime also?

**Comment by hpvb** [comment]
That feature was postponed until 3.1 because recast needed to be modified not to use the STL :)

**Comment by volzhs** [comment]
oh i see

**Comment by SaracenOne** [comment]
Yeah, I wrote about the STL issue in the initial pull request. Still not sure how it can best be resolved. Might even be worth talking to the Recast team themselves since STL is a very small dependency which, honestly, probably doesn't even need to be there.

**Comment by yuyoDev** [comment]
In ArchLinux 64b dont work even with scons platform=android tools=no android_stl=yes

[100%] Compiling ==> thirdparty/recastnavigation/Recast/Source/RecastAlloc.cpp
thirdparty/recastnavigation/Recast/Source/Recast.cpp:26:10: fatal error: 'new' file not found
#include <new>
         ^~~~~
1 error generated.
scons: *** [thirdparty/recastnavigation/Recast/Source/Recast.android.opt.armv7.neon.o] Error 1
scons: building terminated because of errors.

**Comment by AlexHolly** [comment]
Godot Master
It was working ~3 days ago.

Any Idea why I am getting this?

`scons platform=android target=release_debug bits=64 builtin_libpng=yes builtin_openssl=yes builtin_zlib=yes
`
```
In file included from modules/gdnative/pluginscript/pluginscript_instance.cpp:37:
modules/gdnative/pluginscript/pluginscript_script.h:34:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
scons: *** [modules/gdnative/pluginscript/pluginscript_instance.android.opt.debug.64.armv7.neon.o] Error 1
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
@AlexHolly That's a different issue, please file it separately.

**Comment by akien-mga** [comment]
Fixed by #12744.
